gs_chunked_io/writer.py

```diff
@@ -115,11 +115,9 @@ def _compose_dest_blob(self):
             self.bucket.blob(self.key).upload_from_file(io.BytesIO(b""))
 
     def _delete_parts(self, part_names):
-        def _del(name):
-            self.bucket.blob(name).delete()
-
+        blobs = [self.bucket.blob(n) for n in part_names]
```

Comment:
Nit, consider:

```
    def _delete_parts(self, part_names):
        with ThreadPoolExecutor(max_workers=8) as executor:
            for n in part_names:
                executor.submit(_delete_blob, self.bucket.blob(n))
```

Reply:
That would change the effect of the code. `executor.submit`  swallows errors unless futures are collected and `result` called on each one. `map` does this for you.